### ISSUE TITLE
HTML attributes with capital letters are not converted

### ISSUE THREAD

**Issue by tominventisbe** [olifolkerd/tabulator#1742]
**Describe the bug**
Specifying [column configuration](http://tabulator.info/docs/4.1/columns) via HTML `th` elements doesn't seem to work for attributes that contain capitalized characters.

An example of this is the `headerSort` attribute; it seems impossible to set `tabulator-headerSort="false"` via HTML.

The reason seems to be that, even though the attribute is correctly taken from the HTML and inserted into the column definition via `_extractHeaders`, the attribute is - perhaps automatically by the browser, as I don't see any actual conversion happening - converted to lower-case when that happens.

Later on, in the implementation, a check then happens on `column.definition.headerSort !== false`, which then always returns true because the value is `undefined`, whilst the actual value can be found in `column.definition.headersort` due to the casing conversion.

**Tabulator Info**
- Which version of Tabulator are you using? **4.1.5**
- Post a copy of your construct object if possible so we can see how your table is setup

(Roughly reconstructed from an extraction of a much larger example:)

```html
<table>
    <thead>
        <tr>
            <th
                tabulator-field="field"
                tabulator-headerSort="false"
                tabulator-header-sort="false"
            >
                Testcolumn
            </th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>Test</td>
        </tr>
    </tbody>
</table>
```

**To Reproduce**
1. Create a HTML table
2. Specify `tabulator-headerSort` on a column
3. Initialize Tabulator on it
4. The column will still be sortable

**Expected behavior**
The column is not sortable.

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser: Firefox and WebKit(GTK)
 - Version Firefox 64 and WebKitGTK 2.22.5

**Additional context**
This functionality works fine when using it from the JavaScript context.

It is mentioned somewhere that there are limitations to the HTML import, is this perhaps one of them? If so, it may be interesting to explicitly mention this somewhere.

I had trouble earlier where Tabulator was ignoring my `minWidth`, but now I understand that this may be the same issue as that also happens to be a property with capitalization in it, but if this turns out to be another problem, I'll be sure to let you know.

_May_ be related to #1740.

Thanks!

**Comment by dipeshpatel306** [comment]
@tominventisbe  let me know if you found any solution in this same. I also facing same issue.

Thanks 
Dipesh

**Comment by dipeshpatel306** [comment]
@tominventisbe  i changed to variable name cssclass into tabulator.min.js and its working.

**Comment by tominventisbe** [comment]
@dipeshpatel306 @olifolkerd I've produced a workaround solution that may benefit others in the meantime.

```js
const tabulatorHtmlAttributeFixupMap = new Map([
    ['tabulator-headersort', 'tabulator-headerSort'],
    ['tabulator-widthgrow', 'tabulator-widthGrow'],
    ['tabulator-widthshrink', 'tabulator-widthShrink'],
    ['tabulator-minwidth', 'tabulator-minWidth'],
]);

// el = your tabulator element *before* invoking Tabulator itself on it.
el.querySelectorAll('th').forEach((el) => {
    for (const attribute of el.attributes) {
        if (tabulatorHtmlAttributeFixupMap.has(attribute.name)) {
            const newAttribute = document.createAttributeNS('', tabulatorHtmlAttributeFixupMap.get(attribute.name));
            newAttribute.value = attribute.value;

            el.attributes.setNamedItemNS(newAttribute);
        }
    }
});
```

It appears the conversion to lowercase in `element.attributes` [is intentional](https://stackoverflow.com/a/33261208/9788433). To work around it, one has to use a custom XML namespace. I've set that namespace to an empty string here, which appears to work in at least Firefox 65 and a WebKit-based browser.

Because the original attributes no longer seem to contain any reference to their name with its original casing, I found no better solution than to define a map with the properties to rename.

This seems to work for the moment (I can properly disable header sorting now), but I think a better solution for Tabulator would be to use actual data attributes (i.e. change `tabulator-` into `data-tabulator-`), as those properties are automatically converted into their camel-cased equivalents in `element.dataset`, i.e. `data-tabulator-header-sort` will become `element.dataset.tabulatorHeaderSort`.

**Comment by olifolkerd** [comment]
I have pushed a fix for this to the 4.2 branch and will include it in modays release.

Thanks for the headsup

Cheers

Oli :)
